src/Compilers/Core/Portable/CommandLine/CommonCommandLineArguments.cs

```diff
@@ -279,6 +279,40 @@ internal CommandLineArguments()
         {
         }
 
+        internal static string NormalizePathPrefix(string filePath, ImmutableArray<KeyValuePair<string, string>> pathMap)
```

Comment:
I don't see it as coupled with compiler feature. Path Map could be treated as a general self-contained concept.